lib/Doctrine/DBAL/Platforms/OraclePlatform.php

```diff
@@ -418,36 +470,19 @@ protected function _getCreateTableSQL($table, array $columns, array $options = a
      */
     public function getListTableIndexesSQL($table, $currentDatabase = null)
     {
-        $table = $this->normalizeIdentifier($table);
-        $table = $this->quoteStringLiteral($table->getName());
+        return $this->getListIndexesSQL($currentDatabase, $table);
+    }
 
-        return "SELECT uind_col.index_name AS name,
-                       (
-                           SELECT uind.index_type
-                           FROM   user_indexes uind
-                           WHERE  uind.index_name = uind_col.index_name
-                       ) AS type,
-                       decode(
-                           (
-                               SELECT uind.uniqueness
-                               FROM   user_indexes uind
-                               WHERE  uind.index_name = uind_col.index_name
-                           ),
-                           'NONUNIQUE',
-                           0,
-                           'UNIQUE',
-                           1
-                       ) AS is_unique,
-                       uind_col.column_name AS column_name,
-                       uind_col.column_position AS column_pos,
-                       (
-                           SELECT ucon.constraint_type
-                           FROM   user_constraints ucon
-                           WHERE  ucon.index_name = uind_col.index_name
-                       ) AS is_primary
-             FROM      user_ind_columns uind_col
-             WHERE     uind_col.table_name = " . $table . "
-             ORDER BY  uind_col.column_position ASC";
+    /**
+     * Returns the SQL for a list of all indexes in the database.
+     *
+     * @param string $database The database schema. If NULL or '/', the currently active schema will be queried.
+     *
+     * @return string
+     */
+    public function getListAllIndexesSQL(?string $database = null): string
```

Comment:
It should be either nullable or optional, not both. I think the former is preferable.

Reply:
I'll strongly disagree, it should either defaulting to null AND nullable or none. Optionality and nullablility are two things (and for null, the former implies the latter, but not the other way). Missing nullability sign for parameters defaulting to null is PHP's design flaw introduced only because 7.0 had no nullables.